src/handlers/transfers/handler.js

```diff
@@ -335,24 +335,81 @@ const fulfil = async (error, messages) => {
+      }
+
+      // Lets make the error specific if the PayerFSP IDs do not match
+      if (headers[Enum.Http.Headers.FSPIOP.DESTINATION].toLowerCase() !== transfer.payerFsp.toLowerCase()) {
+        fspiopError = ErrorHandler.Factory.createFSPIOPError(ErrorHandler.Enums.FSPIOPErrorCodes.VALIDATION_ERROR, `${Enum.Http.Headers.FSPIOP.DESTINATION} does not match payer fsp on the Fulfil callback response`)
+      }
+
+      const apiFSPIOPError = fspiopError.toApiErrorObject(Config.ERROR_HANDLING)
+
+      // Overriding global boolean declaration with a string value for local as we should handle notifications only to FSPs involved with this transfer
+      const toPayerDestination = transfer.payerFsp
+      const toPayeeDestination = transfer.payeeFsp
+
+      // Set the event details to map to an ABORT_VALIDATION event targeted to the Position Handler
+      const eventDetail = { functionality: TransferEventType.POSITION, action: TransferEventAction.ABORT_VALIDATION }
+
+      // Lets handle the abort validation and change the transfer state to reflect this
+      const transferAbortResult = await TransferService.handlePayeeResponse(transferId, payload, TransferEventAction.ABORT_VALIDATION, apiFSPIOPError)
+
       /**
        * TODO: BULK-Handle at BulkProcessingHandler (not in scope of #967)
        * HOWTO: For regular transfers, send the fulfil from non-payee dfsp.
        * Not sure if it will apply to bulk, as it could/should be captured
        * at BulkPrepareHander. To be verified as part of future story.
        */
-      await Kafka.proceed(Config.KAFKA_CONFIG, params, { consumerCommit, fspiopError: fspiopError.toApiErrorObject(Config.ERROR_HANDLING), eventDetail, fromSwitch, toDestination })
-      throw fspiopError
+
+      // Publish message to Position Handler
+      await Kafka.proceed(Config.KAFKA_CONFIG, params, { consumerCommit, fspiopError: apiFSPIOPError, eventDetail, fromSwitch, toDestination: toPayerDestination })
+
+      /**
+       * Send patch notification callback to original payee fsp if they asked for a a patch response.
+       */
+      if (action === TransferEventAction.RESERVE) {
+        Logger.isInfoEnabled && Logger.info(Util.breadcrumb(location, `callbackReservedAborted--${actionLetter}3`))
+
+        // Set the event details to map to an RESERVE_ABORTED event targeted to the Notification Handler
+        const reserveAbortedEventDetail = { functionality: TransferEventType.NOTIFICATION, action: TransferEventAction.RESERVED_ABORTED }
+
+        // Extract error information
+        const errorCode = apiFSPIOPError && apiFSPIOPError.errorInformation && apiFSPIOPError.errorInformation.errorCode
+        const errorDescription = apiFSPIOPError && apiFSPIOPError.errorInformation && apiFSPIOPError.errorInformation.errorDescription
+
+        // TODO: This should be handled by a PATCH /transfers/{id}/error callback in the future FSPIOP v1.2 specification, and instead we should just send the FSPIOP-Error instead!
```

Comment:
Good to capture this.. maybe we can even add link to this: https://github.com/mojaloop/mojaloop-specification/issues/106